node/src/test_utils/mod.rs

```diff
@@ -174,10 +173,17 @@ impl Waiter {
 }
 
 pub fn make_meaningless_stream_key() -> StreamKey {
```

Comment:
If you moved these two out of here and made them `#[cfg(test)]`-only constructors for `StreamKey`, then you wouldn't have to make the `hash` field of `StreamKey` public, and it could be completely opaque to its clients.